core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -643,6 +643,275 @@ class WorkspaceService(
       }
     }
 
+  def updateWorkspaceBillingProject(workspaceName: WorkspaceName,
+                                    newBillingProjectName: String
+  ): Future[Option[Workspace]] =
+    for {
+      (workspace, destBillingProject) <- validateBillingProjectUpdate(workspaceName, newBillingProjectName)
+      updated <- updateWorkspaceBilling(workspace, destBillingProject)
+    } yield updated
+
+  def validateBillingProjectUpdate(workspaceName: WorkspaceName,
+                                   newBillingProjectName: String
+  ): Future[(Workspace, RawlsBillingProject)] = {
+    val sourceBillingProjectName = RawlsBillingProjectName(workspaceName.namespace)
+    val destBillingProjectName = RawlsBillingProjectName(newBillingProjectName)
+    for {
+      _ <-
+        if (workspaceName.namespace == newBillingProjectName)
+          Future.failed(
+            RawlsExceptionWithErrorReport(
+              ErrorReport(StatusCodes.BadRequest, s"Workspace billing is already set to $newBillingProjectName")
+            )
+          )
+        else Future.unit
+
+      // Source workspace must exist
+      workspaceOpt <- workspaceRepository.getWorkspace(workspaceName)
+      workspace <- workspaceOpt match {
+        case Some(ws) => Future.successful(ws)
+        case None =>
+          Future.failed(
+            RawlsExceptionWithErrorReport(
+              ErrorReport(StatusCodes.NotFound, s"Workspace ${workspaceName.name} does not exist")
+            )
+          )
+      }
```

Comment:
I think you could reduce some of the boilerplate by using `=` instead of `<-`, here and in a few other places in this method, something like this:
```suggestion
      workspaceOpt <- workspaceRepository.getWorkspace(workspaceName)
      workspace = workspaceOpt.getOrElse(
        throw RawlsExceptionWithErrorReport(
          ErrorReport(StatusCodes.NotFound, s"Workspace ${workspaceName.name} does not exist")
        )
      )
```